packages/aws-cdk/lib/util/validate-notification-arn.ts

```diff
@@ -2,5 +2,5 @@
  * Validate SNS topic arn
  */
 export function validateSnsTopicArn(arn: string): boolean {
-  return /^arn:aws:sns:[a-z0-9\-]+:[0-9]+:[a-z0-9\-\_]+$/i.test(arn);
```

Comment:
Could you use [`Arn.split`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.Arn.html#static-splitarn-arnformat) instead? There is still another unsupported partition with this solution (`aws-us-gov`), and it would be cleaner to just test the service and resource name with that method instead of a growing RegExp